refactor: use t.TempDir() instead of os.MkdirTemp

yuhangcangqian opened the pull request:
## Change Description

 TempDir returns a temporary directory for the test to use.The directory is automatically removed when the test and
all its subtests complete. More info  https://pkg.go.dev/testing#B.TempDir

Inspired by https://github.com/btcsuite/btcd/pull/2334

## Steps to Test
Steps for reviewers to follow to test the change.

## Pull Request Checklist
### Testing
- [ ] Your PR passes all CI checks.
- [x] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [ ] The change is not [insubstantial](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#substantial-contributions-only). Typo fixes are not accepted to fight bot spam.
- [ ] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#code-documentation-and-commenting) guidelines, and lines wrap at 80.
- [ ] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#ideal-git-commit-structure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.

📝 Please see our [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

guggero reviewed (commented):
Thanks for the PR.

There are also 2 occurrences in `wtxmgr`.
And it would be nice to get rid of any calls to `os.TempDir()` as well.

yuhangcangqian commented:
> Thanks for the PR.
> 
> There are also 2 occurrences in `wtxmgr`. And it would be nice to get rid of any calls to `os.TempDir()` as well.

Thanks. Modified and rebased. 

Please review again.

guggero commented on wtxmgr/tx_test.go:
```diff
@@ -51,36 +50,28 @@ var (
 		"bdb", filepath.Join(tmpDir, "db"), true, defaultDBTimeout,
 	)
-	return db, func() { os.RemoveAll(tmpDir) }, err
+	return db, func() {}, err
```
If these teardown/cleanup functions are no longer required, let's get rid of them as a return value please.

yuhangcangqian replied:
Thanks for the advice. Modified.
